mitodl/ocw-studio
# add global option to RelationField component

#### Pre-Flight checklist


#### What are the relevant tickets?

closes #780

#### What's this PR do?

this adds a new option to the relation widget which is called `cross_site`. This option basically enables the same 'linking' behavior we have now, but with the added option of picking which website you want to 'look within' in order to find items to link to. This functionality can be used to, for instance, implement resource lists which group together diverse resources from across a swath of websites.

#### How should this be manually tested?

I added a new ocw yaml file based on the one in `ocw-hugo-projects` (by which I mean I copy-pasted). This has a cross-site relation field implementing a resource list set up in it already, so if you set up a new site which uses that configuration you should be able to test it out.

In the UI test out that:

- clicking in to fields you always have some options in the dropdowns
- you see options in the second field after picking a website in the first
- when you add things to the list you can then smoothly reorder them, delete them, etc
- you should be able to save and then re-open things

generally the UI should be unsurprising I think. Hopefully it is!

#### Screenshots (if appropriate)

https://user-images.githubusercontent.com/6207644/142086507-832f1641-e480-4235-b400-5f11cded6771.mov
